src/utp_stream.cpp

```diff
@@ -224,10 +224,10 @@ udp::endpoint utp_socket_impl::remote_endpoint() const
 	return {m_remote_address, m_port};
 }
 
-void utp_socket_impl::send_ack()
+void utp_socket_impl::send_deferred_ack()
 {
-	if (!m_deferred_ack)
-		return;
+	TORRENT_ASSERT(m_deferred_ack);
+	if (!m_deferred_ack) return;
```

Comment:
`if (!m_deferred_ack) return;`

This check might no longer be needed, as we shouldn't be able to reach here without a pending deferred ACK any more.

Reply:
right, hence the assert. I left it in as a precaution